backend/app/domain/model.py

```
@@ -100,6 +122,25 @@ class ModelMappingProviderCreate(ModelMappingProviderBase):
+    billing_mode: BillingMode = Field("token_flat", description="Billing mode")
+    # Per-request fixed price (USD), used when billing_mode == per_request
+    per_request_price: Optional[float] = Field(None, ge=0, description="Per-request price ($)")
+    # Tiered pricing config, used when billing_mode == token_tiered
+    tiered_pricing: Optional[list[TokenTierPrice]] = Field(
+        None, description="Tiered pricing (based on input tokens)"
+    )
+
+    @model_validator(mode="after")
+    def _validate_billing(self) -> "ModelMappingProviderCreate":
+        if self.billing_mode == "per_request" and self.per_request_price is None:
+            raise ValueError("per_request_price is required when billing_mode=per_request")
```

Ruff (0.14.13)

Avoid specifying long messages outside the exception class

(TRY003)